src/components/molecules/Visualizer/Engine/ref.ts

```diff
@@ -4,7 +4,59 @@ import { LatLngHeight, Camera, Typography, Rect } from "@reearth/util/value";
 
 import type { Component } from "../Primitive";
 
+export type MouseEventProps = {
+  x?: number;
+  y?: number;
+  lat?: number;
+  lng?: number;
+  height?: number;
+  layerId?: string;
+  delta?: number;
+};
+
+export type MouseEvents = {
+  click: ((props: MouseEventProps) => void) | undefined;
```

Comment:
Can you choose in a single file either `something?: someType` OR `something: someType | undefined` and stick with it? MouseEventProps uses the first way and then the rest use the second way.

Reply:
Sometimes when i want to express one property is optional i use `something?: somtType` and when i want to express some property exists there but its value can be `undefined` i use `something?: someType | undefined`. But yes these two should be equal and i'll update to use the first one maybe.

Reply:
Sorry i changed this part back as i found there's some difference between the two syntax. When it comes to use `keyof` with the type it will got different result and has effects on other type definitions.